ultralytics/engine/validator.py

```diff
@@ -204,31 +205,43 @@ def __call__(self, trainer=None, model=None):
                 LOGGER.info(f"Results saved to {colorstr('bold', self.save_dir)}")
             return stats
 
-    def match_predictions(self, pred_classes, true_classes, iou):
+    def match_predictions(self, pred_classes, true_classes, iou, use_scipy=False):
         """
         Matches predictions to ground truth objects (pred_classes, true_classes) using IoU.
 
         Args:
             pred_classes (torch.Tensor): Predicted class indices of shape(N,).
             true_classes (torch.Tensor): Target class indices of shape(M,).
             iou (torch.Tensor): An NxM tensor containing the pairwise IoU values for predictions and ground of truth
```

Comment:
Shouldn't it be MxN?

Reply:
Hello, thank you for your comment. You are right, it should be MxN matrix, where M = number of labels, N = number of detections

Reply:
hhh, it seems to be a historical issue.

Reply:
Its just because `box_iou` and `match_predictions` accept first two arguments in reversed order. And comment for `match_predictions` was probably taken from `box_iou` function. That is the whole issue